modules/invenio-oaiserver/invenio_oaiserver/response.py

```diff
@@ -346,25 +347,15 @@ def is_private_index(record):
                             etree.QName(NS_OAIPMH, 'metadata'))
 
     etree_record = copy.deepcopy(record)
+    if not etree_record.get('system_identifier_doi', None):
+        etree_record['system_identifier_doi'] = get_identifier(record)
     if check_correct_system_props_mapping(
-            pid.object_uuid,
+        pid.object_uuid,
             current_app.config.get('OAISERVER_SYSTEM_FILE_MAPPING')):
         etree_record = combine_record_file_urls(etree_record, pid.object_uuid)
 
     root = record_dumper(pid, {'_source': etree_record})
 
-    if check_correct_system_props_mapping(
-            pid.object_uuid,
-            current_app.config.get('OAISERVER_SYSTEM_IDENTIFIER_MAPPING')):
-        if record.pid_doi:
-            root = create_identifier_index(root,
-                                           pid_type=record.pid_doi.pid_type,
-                                           pid_value=record.pid_doi.pid_value)
-        if record.pid_cnri:
-            root = create_identifier_index(root,
-                                           pid_type=record.pid_cnri.pid_type,
-                                           pid_value=record.pid_cnri.pid_value)
-
```

Comment:
Why this you delete this source code?

Reply:
I have changed this method by
etree_record['system_identifier_doi'] = get_identifier(record)